src/Renderer/PointsMaterial.js

```diff
@@ -57,6 +120,73 @@ class PointsMaterial extends THREE.RawShaderMaterial {
         }
     }
 
+    recomputeClassification() {
+        const classification = this.classification;
+        const data = this.classificationTexture.image.data;
+        const width = this.classificationTexture.image.width;
+
+        let valuesChanged = false;
+
+        for (let i = 0; i < width; i++) {
+            let color;
+            let opacity;
+            let visible = true;
+
+            if (classification[i]) {
+                color = classification[i].color;
+                visible = classification[i].visible;
+                opacity = classification[i].opacity;
+            } else if (classification[i % 32]) {
+                color = classification[i % 32].color;
+                visible = classification[i % 32].visible;
+                opacity = classification[i % 32].opacity;
+            } else if (classification.DEFAULT) {
+                color = classification.DEFAULT.color;
+                visible = classification.DEFAULT.visible;
+                opacity = classification.DEFAULT.opacity;
+            } else {
+                color = white;
+                opacity = 1.0;
+            }
+
+            const r = parseInt(255 * color.r, 10);
+            const g = parseInt(255 * color.g, 10);
+            const b = parseInt(255 * color.b, 10);
+            const a = visible ? parseInt(255 * opacity, 10) : 0;
+            const j = 4 * i;
+
+            if (data[j + 0] !== r) {
+                data[j + 0] = r;
+                valuesChanged = true;
+            }
```

Comment:
Is it really necessary to do all of this ? As far as I know, this shouldn't be done a lot of time, so maybe you could simplify the code here ?

Reply:
simplified